exporter/prometheusexporter/accumulator.go

```diff
@@ -332,3 +336,80 @@ func createMetric(metric pmetric.Metric) pmetric.Metric {
 
 	return m
 }
+
+func AccumulateNumericValues(prev, current, dest pmetric.NumberDataPoint) {
+	if current.StartTimestamp().AsTime().Before(prev.StartTimestamp().AsTime()) {
+		dest.SetStartTimestamp(current.StartTimestamp())
+	} else {
+		dest.SetStartTimestamp(prev.StartTimestamp())
+	}
+
+	older := prev
+	newer := current
+	if current.Timestamp().AsTime().Before(prev.Timestamp().AsTime()) {
+		older = current
+		newer = prev
+	}
+
+	newer.Attributes().CopyTo(dest.Attributes())
+	dest.SetTimestamp(newer.Timestamp())
+
+	switch newer.ValueType() {
+	case pmetric.NumberDataPointValueTypeDouble:
+		dest.SetDoubleVal(newer.DoubleVal() + older.DoubleVal())
+	case pmetric.NumberDataPointValueTypeInt:
+		dest.SetIntVal(newer.IntVal() + older.IntVal())
+	}
+}
+
+func AccumulateHistogramValues(prev, current, dest pmetric.HistogramDataPoint) {
+	if current.StartTimestamp().AsTime().Before(prev.StartTimestamp().AsTime()) {
+		dest.SetStartTimestamp(current.StartTimestamp())
+	} else {
+		dest.SetStartTimestamp(prev.StartTimestamp())
+	}
+
+	older := prev
+	newer := current
+	if current.Timestamp().AsTime().Before(prev.Timestamp().AsTime()) {
+		older = current
+		newer = prev
+	}
+
+	newer.Attributes().CopyTo(dest.Attributes())
+	dest.SetTimestamp(newer.Timestamp())
+
+	match := true
+	if older.ExplicitBounds().Len() == newer.ExplicitBounds().Len() {
+		for i := 0; i < newer.BucketCounts().Len(); i++ {
+			if older.ExplicitBounds().At(i) != newer.ExplicitBounds().At(i) {
+				match = false
```

Comment:
```suggestion
				match = false
				break
```

No point checking any further values once we know they don't match.